vowpalwabbit/cb_adf.cc

```diff
@@ -280,54 +278,48 @@ void output_rank_example(vw& all, cb_adf& c, example& ec, v_array<example*>* ec_
   CB::print_update(all, is_test, ec, ec_seq, true);
 }
 
-void output_example_seq(vw& all, cb_adf& data)
+void output_example_seq(vw& all, cb_adf& data, multi_ex& ec_seq)
 {
-  if (data.ec_seq.size() > 0)
+  if (ec_seq.size() > 0)
   {
     if (data.rank_all)
-      output_rank_example(all, data, **(data.ec_seq.begin()), &(data.ec_seq));
+      output_rank_example(all, data, **(ec_seq.begin()), &(ec_seq));
     else
     {
-      output_example(all, data, **(data.ec_seq.begin()), &(data.ec_seq));
+      output_example(all, data, **(ec_seq.begin()), &(ec_seq));
 
       if (all.raw_prediction > 0)
-        all.print_text(all.raw_prediction, "", data.ec_seq[0]->tag);
+        all.print_text(all.raw_prediction, "", ec_seq[0]->tag);
     }
   }
 }
 
-void clear_seq_and_finish_examples(vw& all, cb_adf& data)
+void clear_seq_and_finish_examples(vw& all, multi_ex& ec_seq)
 {
-  if (data.ec_seq.size() > 0)
-    for (example* ecc : data.ec_seq)
+  if (ec_seq.size() > 0)
+    for (example* ecc : ec_seq)
       if (ecc->in_use)
         VW::finish_example(all, ecc);
-  data.ec_seq.erase();
+  ec_seq.erase();
 }
 
-void finish_multiline_example(vw& all, cb_adf& data, example& ec)
+void finish_multiline_example(vw& all, cb_adf& data, multi_ex& ec_seq)
 {
-  if (data.need_to_clear)
+  if (ec_seq.size() > 0)
   {
-    if (data.ec_seq.size() > 0)
-    {
-      output_example_seq(all, data);
-      global_print_newline(all);
-    }
-    clear_seq_and_finish_examples(all, data);
-    data.need_to_clear = false;
+    output_example_seq(all, data, ec_seq);
+    global_print_newline(all);
   }
+  clear_seq_and_finish_examples(all, ec_seq);
 }
 
 void end_examples(cb_adf& data)
 {
-  if (data.need_to_clear)
-    data.ec_seq.erase();
+  // noop
```

Comment:
can't you remove this completely and then simply not register the callback?

Reply:
Yes. removed.